test/AffineOps/canonicalize.mlir

```diff
@@ -424,3 +424,33 @@ func @fold_empty_loop() {
   }
   return
 }
+
+// -----
+
+// CHECK-DAG: [[LBMAP:#map[0-9]+]] = ()[s0] -> (0, s0)
+// CHECK-DAG: [[UBMAP:#map[0-9]+]] = ()[s0] -> (1024, s0 + s0)
```

Comment:
shouldn't this fold to `2 * s0` ?

Reply:
We didn't include such a simplification (or even say s0 - s0) in AffineExpr::simplifyAdd; so this won't fold. I think AffineExpr::simplifyAdd should be updated to at least fold diff of identical expressions by construction.